Popup/tests/Popup-specs.js

```diff
@@ -1,180 +1,181 @@
+					<Popup centered open />
+				</FloatingLayerController>
 			);
 
-			const expected = true;
-			const actual = popup.prop('centered');
+			const expected = 'centered';
+			const popup = screen.getByRole('alert');
 
-			expect(actual).toBe(expected);
+			expect(popup).toHaveClass(expected);
 		});
 
 		test('should not be rendered with centered content if centered is set to false', () => {
-			const popup = mount(
-				<Popup centered={false} />
+			render(
+				<FloatingLayerController>
+					<Popup open />
+				</FloatingLayerController>
 			);
 
-			const expected = false;
-			const actual = popup.prop('centered');
+			const expected = 'centered';
+			const popup = screen.getByRole('alert');
 
-			expect(actual).toBe(expected);
+			expect(popup).not.toHaveClass(expected);
 		});
 	});
 
 	describe('with position center', function () {
 		test('should have position=center when no position is specified', () => {
-			const popup = mount(
-				<PopupBase />
+			render(
+				<FloatingLayerController>
+					<Popup open />
+				</FloatingLayerController>
 			);
 
 			const expected = 'center';
-			const actual = popup.prop('position');
+			const popup = screen.getByRole('alert');
 
-			expect(actual).toBe(expected);
+			expect(popup).toHaveClass(expected);
 		});
 	});
 
 	describe('with position top', function () {
 		test('should have top class', () => {
-			const popup = shallow(
-				<PopupBase open position="top">
-					<div>popup</div>
-				</PopupBase>
+			render(
+				<FloatingLayerController>
+					<Popup open position="top"><div>popup</div></Popup>
+				</FloatingLayerController>
 			);
 
-			expect(popup.find(`.${css.popup}`).prop('className').split(' ')).toContain('top');
+			const expected = 'top';
+			const popup = screen.getByRole('alert');
+
+			expect(popup).toHaveClass(expected);
 		});
 	});
 
 	describe('with position changes dynamically', function () {
 		test('should not have top class when position change from top to any other position', () => {
-			const firstPosition = 'top';
-			const popup = shallow(
-				<PopupBase open position={firstPosition}>
-					<div>popup</div>
-				</PopupBase>
+			const {rerender} = render(
+				<FloatingLayerController>
+					<Popup open position="top"><div>popup</div></Popup>
+				</FloatingLayerController>
 			);
 
-			expect(popup.find(`.${css.popup}`).prop('className').split(' ')).toContain(firstPosition);
-			expect(popup.find(`.${css.popup}`).prop('className').split(' ')).not.toContain('center');
-			expect(popup.prop('className').split(' ')).not.toContain('center');
+			const firstExpected = 'top';
+			const initialPopup = screen.getByRole('alert');
 
-			popup.setProps({position: 'center'});
-
-			expect(popup.find(`.${css.popup}`).prop('className').split(' ')).not.toContain(firstPosition);
-			expect(popup.prop('className').split(' ')).not.toContain(firstPosition);
-		});
-	});
-
-	test('should have `center` class when position prop is set to `center` (default)', () => {
```

Comment:
Why did you overwrite the "should have `center` class" and "should have `top` class" tests?

Reply:
The popup with position=center and position=top were tested above. But I will add them back.